packages/apollo-server-client/index.js

```diff
@@ -0,0 +1,242 @@
+const { ApolloClient, InMemoryCache, ApolloLink } = require('@apollo/client')
+const { BatchHttpLink } = require('@apollo/client/link/batch-http')
+const { createUploadLink } = require('apollo-upload-client')
+const { onError }  = require('apollo-link-error')
+const { RetryLink } = require('@apollo/client/link/retry')
+const { getLogger } = require('@condo/keystone/logging')
+const fetch = require('cross-fetch/polyfill').fetch
+const { chunk: splitArray } = require('lodash')
+
+const { MAX_REQUESTS_IN_BATCH, MAX_MODIFY_OPERATIONS_IN_REQUEST, MAX_RETRIES_ON_NETWORK_ERROR, LOAD_CHUNK_SIZE } = require('./constants')
+const { SIGNIN_BY_EMAIL_MUTATION, SIGNIN_BY_PHONE_AND_PASSWORD_MUTATION } = require('./lib/gql')
+
+class ApolloServerClient {
+
+    client
+    clientName
+    authToken
+    userId
+    locale
+    endpoint
+    authRequisites = {}
+    logger
+
+    /**
+     *
+     * @param endpoint - https://condo.d.doma.ai/admin/api
+     * @param authRequisites - can be { identity: 'service-user@doma.ai', secret: 'password' } or { phone: '+7911....', password: '' }
+     * @param clientName - logger name
+     * @param locale - for server side translated texts
+     */
+    constructor (endpoint, authRequisites = {}, { clientName = 'apollo-server-client', locale = 'ru' } = {}) {
+        this.clientName = clientName
+        this.endpoint = endpoint
+        this.authRequisites = authRequisites
+        this.logger = getLogger(clientName)
+        this.batchClient = this.createClient([this.errorLink(), this.authLink(), this.retryLink(), this.batchTerminateLink()])
+        this.client = this.createClient([this.errorLink(), this.authLink(), this.retryLink(), this.uploadTerminateLink()])
+    }
+
+    dvSender () {
+        return { dv: 1, sender: { dv: 1, fingerprint: this.clientName } }
+    }
+
+    createClient (links) {
```

Comment:
Can it be done in constructor?

Reply:
It creates 2 clients
1 with uploadLink as terminated
2 with batch link as terminated